posthog/client.py

```diff
@@ -354,39 +354,38 @@ def feature_enabled(self, key, distinct_id, default=False, *, groups={}):
         require("distinct_id", distinct_id, ID_TYPES)
         require("groups", groups, dict)
 
-        if not self.personal_api_key:
-            self.log.warning("[FEATURE FLAGS] You have to specify a personal_api_key to use feature flags.")
-        if not self.feature_flags:
+        if self.feature_flags == None and self.personal_api_key:
             self.load_feature_flags()
+        response = None
 
         # If loading in previous line failed
-        if not self.feature_flags:
-            response = default
-        else:
+        if self.feature_flags:
             for flag in self.feature_flags:
                 if flag["key"] == key:
                     feature_flag = flag
-                    break
-            else:
-                return default
-
-            if feature_flag.get("is_simple_flag"):
-                response = _hash(key, distinct_id) <= ((feature_flag.get("rollout_percentage", 100) or 100) / 100)
+                    if feature_flag.get("is_simple_flag"):
+                        response = _hash(key, distinct_id) <= (
+                            (feature_flag.get("rollout_percentage", 100) or 100) / 100
+                        )
+        if not response:
```

Comment:
If it's a simple flag, yet doesn't match (response is `False`), won't we make a needless `decide` call? Shouldn't this compare with `None` instead?

Reply:
Yep good point, will test